src/Conversion/KrnlToLLVM/KrnlInstrument.cpp

```diff
@@ -72,10 +72,11 @@ class KrnlInstrumentOpLowering : public ConversionPattern {
         if (auto nameLocIt = locIt.dyn_cast<NameLoc>())
           name += nameLocIt.getName().str() + "-";
         else if (auto fileLineColLoc = locIt.dyn_cast<FileLineColLoc>()) {
-          StringRef filename =
-              llvm::sys::path::filename(fileLineColLoc.getFilename().str());
-          name += filename.str() + ":" +
-                  std::to_string(fileLineColLoc.getLine()) + "-";
+          std::string filename =
+              llvm::sys::path::filename(fileLineColLoc.getFilename().str())
+                  .str();
+          name +=
+              filename + ":" + std::to_string(fileLineColLoc.getLine()) + "-";
```

Comment:
alternatively
```
StringRef filename = llvm::sys::path::filename(fileLineColLoc.getFilename().strref());
```
would result in a stringref pointing into the `fileLineColLoc.getFilename()` StringAttr which should work (and be more efficient although that doesn't matter in this context)

Reply:
@sorenlassen Thanks for your comments! I found this new code works through trial and error, but honestly speaking, I'm still not sure why old code didn't work, but new code worked. 

From your alternative code, should I use `llvm::sys::path::filename(fileLineColLoc.getFilename().strref()) ` also in new code?

Reply:
the old code didn't work because `fileLineColLoc.getFilename().str()` created a temporary string object with a lifetime until the end of the expression
```
path::filename(fileLineColLoc.getFilename().str())
```
and `path::filename(..)` returned a StringRef pointing into the temporary object (StringRef points into memory that it doesn't own, like ArrayRef or a C-style `char*` string)

the new code
```
path::filename(fileLineColLoc.getFilename().str()).str()
```
works because the final `.str()` constructs a string object with a copy of the StringRef memory returned by `path::filename(..)` before the end of the lifetime of the innermost temporary object (its lifetime extends to the end of the lifetime of the enclosing expression)

Reply:
Thanks for the details! I understood well.